tools/compileProtos.ts

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env node
+
+/*
+ * Copyright 2019 Google LLC
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as util from 'util';
+import * as pbjs from 'protobufjs/cli/pbjs';
+
+const readdir = util.promisify(fs.readdir);
+const readFile = util.promisify(fs.readFile);
+const stat = util.promisify(fs.stat);
+const pbjsMain = util.promisify(pbjs.main);
+
+const PROTO_LIST_REGEX = /_proto_list.json$/;
+
+/**
+ * Recursively scans directories starting from `directory` and finds all files
+ * matching `PROTO_LIST_REGEX`.
+ *
+ * @param {string} directory Path to start the scan from.
+ * @return {Promise<string[]} Resolves to an array of strings, each element is a full path to a matching file.
+ */
+async function findProtoJsonFiles(directory: string): Promise<string[]> {
```

Comment:
I feel like this code already exists somewhere.
https://github.com/googleapis/nodejs-proto-files/blob/master/src/load.ts#L39

Can't remember if `walkdir` is brought in at runtime, but it seems like something we already use.

Reply:
We won't need that code and `walkdir` (but removing that part will be a breaking change). Here, I preferred I just do it manually without using any extra dependency in this script, other than `protobufjs`, obviously.

Reply:
And yes, we won't actually need `nodejs-proto-files` either.